apps/dashboard/src/hooks/use-form-autosave.ts

```diff
@@ -14,19 +15,42 @@ export const useFormAutoSave = <T extends FieldValues>({
 }) => {
   const onSubmitRef = useDataRef(onSubmit);
   const { formState, control, handleSubmit } = form;
+  const previousStepsLength = useRef<number | null>(null);
 
   const watchedData = useWatch<T>({
     control,
   });
 
-  const debouncedSave = useDebounce(() => {
+  const save = () => {
     if (enabled) {
       handleSubmit(onSubmitRef.current)();
     }
-  }, 500);
+  };
+
+  const debouncedSave = useDebounce(save, 500);
+
+  const checkStepsDeleted = useCallback(() => {
+    const currentStepsLength = watchedData.steps?.length ?? 0;
+    const wasDeleted = previousStepsLength.current !== null && currentStepsLength < previousStepsLength.current;
+
+    previousStepsLength.current = currentStepsLength;
+
+    return wasDeleted;
+  }, [watchedData]);
```

Comment:
Logic to determine if step was deleted in order to not use debounced save.

If we used debounced save for deletion, user could potentially refresh in the 500ms time window and the step wouldn't be deleted, although I'm not sure if its worth the additional logic.